File: unimagtienda/src/main/java/com/tienda/unimagtienda/app/customer/entity/Customer.java
package com.tienda.unimagtienda.app.customer.entity;

import com.tienda.unimagtienda.app.address.entity.Address;
import com.tienda.unimagtienda.app.customer.enums.CustomerStatus;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "customers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    private String email;

    @Enumerated(EnumType.STRING)
    private CustomerStatus status = CustomerStatus.ACTIVE;


    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Address> addresses = new ArrayList<>();


}

File: fronend/ecomerce/src/features/customer/types/customer.ts
export type CustomerStatus = "ACTIVE" | "INACTIVE";

export interface CustomerResponse {
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    status: CustomerStatus;
}
export interface UpdateCustomerRequest {
    firstName: string;
    lastName: string;
    email: string;
    status: CustomerStatus;
}
export interface CreateCustomerRequest {
    firstName: string;
    lastName: string;
    email: string;
}